src/EnergyPlus/UnitarySystem.cc

```diff
@@ -13268,22 +13268,24 @@ namespace UnitarySystems {
                                                                                                       singleMode);
                             if ((state.dataLoopNodes->Node(OutletNode).HumRat - DesOutHumRat) < Acc) break;
                         }
+                        // make sure outlet HumRat is below set point before calling SolveRoot
+                        if (state.dataLoopNodes->Node(OutletNode).HumRat < DesOutHumRat) {
```

Comment:
@rraustad there are a few trivial things to clean up on this PR that I'd be happy to tackle.  This comment looks to be the only real technical question that would need a response.  Do you have any thoughts on it?  I'm looking to wrap up a bunch of dangling PRs and I'm happy to shepherd this one toward the merge queue if you can just comment on this.  Thanks!

Reply:
Yeah, these different coil models do different things during the sensible iteration part of the code (e.g., setting this->m_CoolingSpeedRatio or not). For CoolingDX, the sensible calc uses this->m_CoolingSpeedRatio = 1 and then iterates on PLR, if needed, but does not yet set this->m_CoolingSpeedRatio = PLR until the end of this function (via this->m_CoolingSpeedRatio = SpeedRatio) here:

        this->m_CoolingPartLoadFrac = PartLoadFrac;
        this->m_CoolingSpeedRatio = SpeedRatio;
        this->m_CoolingCycRatio = CycRatio;
        this->m_DehumidificationMode = DehumidMode;

So that when we get to the humidity control section here, this->m_CoolingSpeedRatio still = 1, and PartLoadFrac is set to 1 (in case it was lowered during the sensible calc), so here we are looping through the speeds at full compressor speed, for that speed, until we find the speed that exceeds the humidity ratio set point. And then, if that speed's outlet humidity ratio is less than the set point (otherwise we would be at max speed and couldn't get there) call RootSolver and find the new PLR, and also set SpeedRatio = PLR after calling RootSolver. For the case where "say speedNum = 2 because it nailed DesOutHumRat" then SpeedRatio should = 1 because that is what was used when it "nailed it". I ran through this again just now using the new unit test. I don't see any logic flaw but another review by @mjwitte with this explaination in hand couldn't hurt.

Reply:
Well, I do see a logic flaw. In the humidity control code for DehumCtrlType::Multimode (above here) the RootSolver call is not protected as it is here. These should look identical. @Myoldmopar let me know is this is becomming less trivial.